aws/resource_aws_sagemaker_code_repository.go

```diff
@@ -0,0 +1,205 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsSagemakerCodeRepository() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSagemakerCodeRepositoryCreate,
+		Read:   resourceAwsSagemakerCodeRepositoryRead,
+		Update: resourceAwsSagemakerCodeRepositoryUpdate,
+		Delete: resourceAwsSagemakerCodeRepositoryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"code_repository_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 63),
+			},
+			"git_config": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"repository_url": {
+							Type:     schema.TypeString,
+							ForceNew: true,
+							Required: true,
+						},
+						"branch": {
+							Type:         schema.TypeString,
+							ForceNew:     true,
+							Optional:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1024),
+						},
+						"secret_arn": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateArn,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsSagemakerCodeRepositoryCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	name := d.Get("code_repository_name").(string)
+
+	createOpts := &sagemaker.CreateCodeRepositoryInput{
+		CodeRepositoryName: aws.String(name),
+		GitConfig:          expandSagemakerCodeRepositoryGitConfig(d.Get("git_config").([]interface{})),
+	}
+
+	log.Printf("[DEBUG] sagemaker code repository create config: %#v", *createOpts)
+	_, err := conn.CreateCodeRepository(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating Sagemaker code repository: %w", err)
+	}
+
+	d.SetId(name)
+
+	return resourceAwsSagemakerCodeRepositoryRead(d, meta)
+}
+
+func resourceAwsSagemakerCodeRepositoryRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	describeNotebookInput := &sagemaker.DescribeCodeRepositoryInput{
+		CodeRepositoryName: aws.String(d.Id()),
+	}
+	CodeRepository, err := conn.DescribeCodeRepository(describeNotebookInput)
```

Comment:
```suggestion
	input := &sagemaker.DescribeCodeRepositoryInput{
		CodeRepositoryName: aws.String(d.Id()),
	}

	codeRepository, err := conn.DescribeCodeRepository(describeNotebookInput)
```

and change `CodeRepository` to `codeRepository` below.

Reply:
Also, maybe move to an internal `finder` package (and reuse that function in `testAccCheckAWSSagemakerCodeRepositoryDestroy` and `testAccCheckAWSSagemakerCodeRepositoryExists`)?

Reply:
added